PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalInternalClient.kt

```diff
@@ -247,19 +253,28 @@ internal class PayPalInternalClient(
 
     private fun getClientMetadataId(
         context: Context,
+        correlationId: String?,
         callback: (String?) -> Unit
     ) {
         braintreeClient.getConfiguration { configuration, error ->
             if (error != null) {
                 callback(error("No Client Metadata Id"))
             } else {
-                val clientMetadataId = dataCollector.getClientMetadataId(
-                    context,
-                    configuration,
+                val request = DataCollectorRequest(false, correlationId)
+
+                val dataCollectorRequest = DataCollectorInternalRequest(
                     false
+                ).apply {
+                    applicationGuid = dataCollector.getPayPalInstallationGUID(context)
```

Comment:
Do we need to be doing this apply on the `applicationGuid`? I am not sure the context of what this does on Android

Reply:
The existing PayPal is doing the same thing as we're doing here. Also, this function seems to be updating state for both the DataCollector and BraintreeSharedPreferences.